server/shared/src/main/scala/org/http4s/server/middleware/LoggerConfig.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package server
+package middleware
+
+import cats.effect.kernel.Async
+import org.typelevel.ci.CIString
+
+/** Configuration for logging actions in server middleware.
+  *
+  * @param logHeaders Whether to log headers
+  * @param logBody Whether to log the body
+  * @param redactHeadersWhen Function to determine which headers should be redacted
+  * @param logAction The effectful log action that returns a logging function
+  */
+sealed abstract case class LoggerConfig[F[_]] private[middleware] (
```

Comment:
Would it be problematic to extend it in a binary-compatible way in the future? 
Even though the case class is sealed.

https://mimalyzer.indoorvivants.com/comparison/a134a982-9f94-4f36-8d8f-7f7b05a0c270

Reply:
I've changed it to a `sealed abstract class` and mimalyzer seems happy. New version: https://mimalyzer.indoorvivants.com/comparison/e8f82479-11fd-4678-a1f3-adc8f92f3723

(Handy little tool, I hadn't come across it! Is there a way to validate that sort of thing via the sbt plugin?)

I am not sure if there is a simpler approach for extending these builders in a binary-compatible way and would be happy to simplify it